lib/checks/resolvesToConstant.js

```diff
@@ -1,24 +1,21 @@
-const Node = require('../node');
+const {query} = require('math-nodes');
 
 // Returns true if the node is a constant or can eventually be resolved to
 // a constant.
 // e.g. 2, 2+4, (2+4)^2 would all return true. x + 4 would return false
 function resolvesToConstant(node) {
-  if (Node.Type.isOperator(node) || Node.Type.isFunction(node)) {
+  if (query.isApply(node)) {
     return node.args.every(
       (child) => resolvesToConstant(child));
   }
-  else if (Node.Type.isParenthesis(node)) {
-    return resolvesToConstant(node.content);
-  }
-  else if (Node.Type.isConstant(node, true)) {
+  else if (query.isNumber(node)) {
     return true;
   }
-  else if (Node.Type.isSymbol(node)) {
+  else if (query.isIdentifier(node)) {
     return false;
   }
-  else if (Node.Type.isUnaryMinus(node)) {
-    return resolvesToConstant(node.args[0]);
+  else if (node.type === 'Parentheses') {
```

Comment:
how do you differentiate between cases where you could flatten x + y + x, and cases where it should be kept as (x + y) + z for pedagogical reasons? Or do you think we could make it work by parsing the tree to be flat and then not trying to flatten the tree at all in the process of solving the problem?